src/heritedcode_api_functions.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import requests
+import json
+
+
+class ApiResponse:
+    """
+    this class represents response data structure for returning response
+    from functions.
+    """
+
+    def __init__(
+        self,
+        checksums,
+        data_url,
+        filetype_url,
+        language_url,
+        length,
+        license_url,
+        status,
+    ):
+        self.checksums = checksums
+        self.data_url = data_url
+        self.filetype_url = filetype_url
+        self.language_url = language_url
+        self.length = length
+        self.license_url = license_url
+        self.status = status
+
+    def __eq__(self, second_class_obj):
+        if isinstance(second_class_obj, ApiResponse):
+            for key in vars(self):
+                if getattr(self, key) != getattr(second_class_obj, key):
+                    return False
+            return True
+        return False
+
+    """
+    Returns an instance of ApiResponse class.
+    """
+
+    @classmethod
```

Comment:
How is it different from init function ?

Reply:
Actually @pombredanne asked to keep a @classmethod and genrate instance of class from there only.